rviz_default_plugins/src/rviz_default_plugins/displays/marker/marker_common.cpp

```diff
@@ -153,8 +154,32 @@ void MarkerCommon::addMessage(const visualization_msgs::msg::Marker::ConstShared
 void MarkerCommon::addMessage(
   const visualization_msgs::msg::MarkerArray::ConstSharedPtr array)
 {
+  using ns_type = decltype(visualization_msgs::msg::Marker::ns);
+  using id_type = decltype(visualization_msgs::msg::Marker::id);
+  using pair_type = std::pair<id_type, const ns_type &>;
+
+  // Keep track of unique markers
+  std::set<pair_type> unique_markers;
+  bool found_duplicate = false;
+
+  std::unique_lock<std::mutex> lock(queue_mutex_);
```

Comment:
nit (another small performance question) Should you defer obtaining the lock until after you have checked for duplicates? 
Just in case holding for extra 4ms (in 10000 marker case) may bottleneck other operations.

e.g.
```cc
// First add messages for display.
{
  lock(queue_mutex_);
  for (marker : markers) {
    addMessage(...);
  }
}
// Scan markers for duplicates.
...
```

Reply:
thanks!